### ISSUE TITLE
ResourceExpired for mega.nz storage

### ISSUE THREAD

**Issue by nikolayhg** [duplicati/duplicati#3773]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.4.5_beta_2018-11-28
- **Operating system**: Ubuntu 18 
- **Backend**: mega.nz

## Description

When testing the connection to mega.nz I get the error "ResourceExpired"

## Steps to reproduce

1. Add a backup with the storage type mega.nz
2. Provide your login and test the connection

- **Actual result**:
error
- **Expected result**:
success

## Debug log

```
CG.Web.MegaApiClient.ApiException: API response: ResourceExpired
  at CG.Web.MegaApiClient.MegaApiClient.RequestCore[TResponse] (CG.Web.MegaApiClient.Serialization.RequestBase request, System.Byte[] key) [0x00132] in <b77b6a46b6b94e008d0d458c52698090>:0 
  at CG.Web.MegaApiClient.MegaApiClient.Request[TResponse] (CG.Web.MegaApiClient.Serialization.RequestBase request, System.Byte[] key) [0x00024] in <b77b6a46b6b94e008d0d458c52698090>:0 
  at CG.Web.MegaApiClient.MegaApiClient.Login (CG.Web.MegaApiClient.MegaApiClient+AuthInfos authInfos) [0x00037] in <b77b6a46b6b94e008d0d458c52698090>:0 
  at CG.Web.MegaApiClient.MegaApiClient.Login (System.String email, System.String password) [0x00009] in <b77b6a46b6b94e008d0d458c52698090>:0 
  at Duplicati.Library.Backend.Mega.MegaBackend.get_Client () [0x0000e] in <b4f95326982b44b59db3ece5ce06b53b>:0 
  at Duplicati.Library.Backend.Mega.MegaBackend.GetCurrentFolder (System.Boolean autocreate) [0x00020] in <b4f95326982b44b59db3ece5ce06b53b>:0 
  at Duplicati.Library.Backend.Mega.MegaBackend.ResetFileCache (System.Collections.Generic.IEnumerable`1[T] list) [0x00008] in <b4f95326982b44b59db3ece5ce06b53b>:0 
  at Duplicati.Library.Backend.Mega.MegaBackend.List () [0x00008] in <b4f95326982b44b59db3ece5ce06b53b>:0 
  at Duplicati.Library.Interface.BackendExtensions.TestList (Duplicati.Library.Interface.IBackend backend) [0x00000] in <cad0aa948bd5428e9683582f0d34e521>:0 
  at Duplicati.Library.Backend.Mega.MegaBackend.Test () [0x00000] in <b4f95326982b44b59db3ece5ce06b53b>:0 
  at Duplicati.Server.WebServer.RESTMethods.RemoteOperation.TestConnection (System.String url, Duplicati.Server.WebServer.RESTMethods.RequestInfo info) [0x000b7] in <fe28905ee30b422e8d475f1cfdb85515>:0 
  at Duplicati.Server.WebServer.RESTMethods.RemoteOperation.POST (System.String key, Duplicati.Server.WebServer.RESTMethods.RequestInfo info) [0x00091] in <fe28905ee30b422e8d475f1cfdb85515>:0 
  at Duplicati.Server.WebServer.RESTHandler.DoProcess (Duplicati.Server.WebServer.RESTMethods.RequestInfo info, System.String method, System.String module, System.String key) [0x00280] in <fe28905ee30b422e8d475f1cfdb85515>:0 
```

**Comment by warwickmm** [comment]
I don't have any issues connecting to Mega.nz and running backups.  Do you have any uncommon characters in your username or password that may possibly be causing some issues?

**Comment by nikolayhg** [comment]
the password contains few common special characters. No uncommon chars.
I am able to login with same login on the mega.nz website.

**Comment by warwickmm** [comment]
This sounds similar to 

https://forum.duplicati.com/t/mega-nz-error-when-check-connection/5316

In version 2.0.4.12, we upgraded the mega client to the latest version.  I'm not sure if that fixes the issue though.  Are you willing to test the newest Canary version?  Note that the Canary versions are more bleeding-edge and can have other issues and instabilities.

**Comment by nikolayhg** [comment]
thanks for the info! Can't install new version now. If I manage to do it in the next weeks will share the result.

**Comment by jazzl0ver** [comment]
```Duplicati - 2.0.4.23_beta_2019-07-14``` - still having the issue

**Comment by warwickmm** [comment]
2.0.4.23 does not have the latest mega client version.  2.0.4.23 is the same as 2.0.4.5, but with a warning for Amazon Cloud Drive users about a pending shutdown.

**Comment by individuell** [comment]
I'm having the same issue on version `2.0.4.23_beta_2019-07-14`. The only thing that is worth mentioning is, that my email address contains a plus sign.

**Comment by kastmada** [comment]
...same problem here. It seems like the problem is only about the new accounts. Just opened fresh Mega account and it gives the same error as described here. When tried on old mega accounts it works without any issues.

**Comment by ts678** [comment]
> It seems like the problem is only about the new accounts.

Thanks, but it's old news. A new accounts issue was probably fixed in MegaApiClient [v1.7.0](https://github.com/gpailler/MegaApiClient/releases/tag/v1.7.0):

> `#108 / #110` Added support for new Mega accounts (using new authentication system).

[Mega.nz error when check connection](https://forum.duplicati.com/t/mega-nz-error-when-check-connection/5316/4?u=ts678)

> Old Mega accounts still working. It’s only happening with new accounts

[upgraded mega api to 1.7.1 #3593](https://github.com/duplicati/duplicati/pull/3593) newer than needed for the fix, but picked up another too.

[v2.0.4.11-2.0.4.11_canary_2019-01-16](https://github.com/duplicati/duplicati/releases/tag/v2.0.4.11-2.0.4.11_canary_2019-01-16)

> Updated the Mega API, thanks @Pectojin

and unfortunately it's not in any beta because the latest beta code base lacks a year of fixes.
There are definitely people working to get it out, but stabilizing it for release has been hard.

If someone is willing to try a Canary release (sometimes buggy), we'll see if it fixes this issue.

[API response: ResourceExpired #110](https://github.com/gpailler/MegaApiClient/issues/110)

was also a ResourceExpired, so my guess would be that the fixed library will fix Duplicati too.

**Comment by individuell** [comment]
I've mingled around with my Duplicati setup and it's version and suddenly it worked. Unsure why.

I assume it had to do something with my setup not installing the latest version, thus using outdated software which didn't actually contain the fix.

I used chocolatey package manager to install Duplicati. Switching between release branches seemed to work but actually didn't (I assume).

The solution was to (backup profile and) uninstall Duplicati entirely, then download the latest version containing the fixes and the problem was not existing anymore with freshly created accounts.

Good luck
